src/Models/UnderDevelopment/qsm_sb.m

```diff
@@ -0,0 +1,431 @@
+      disp('-----------------------------------------------');
+
+    end
+
+  elseif FitOpt.regL2_Flag && not(FitOpt.reoptL2_Flag ) % || DO NOT reopt Lambda L2 case chi_L2 generation
+
+    if isempty(FitOpt.LambdaL2) % In case user forgets
+
+      error('Lambda2 value is needed. Please select Re-opt LambdaL2 if you dont know the value');
+
+    else
+
+      disp('Skipping reoptimization of Lambda L2.');
+      lambdaL2 = FitOpt.LambdaL2;
+
+    end
+
+    if FitOpt.magnW_Flag % MagnitudeWeighting is present | Lambdal2 known
+
+      disp('Started   : Calculation of chi_L2 map with magnitude weighting...');
+      [FitResults.chiL2,FitResults.chiL2pcg] = calcChiL2(phaseLUnwrap, lambdaL2, FitOpt.direction, imageResolution, maskGlobal, padSize, magnWeight);
+      disp('Completed   : Calculation of chi_L2 map with magnitude weighting.');
+      disp('-----------------------------------------------');
+
+    else % magn weight is not present | Lambdal2 known
+
+      disp('Started   : Calculation of chi_L2 map without magnitude weighting...');
+      [FitResults.chiL2] = calcChiL2(phaseLUnwrap, lambdaL2, FitOpt.direction, imageResolution, maskGlobal, padSize);
+      disp('Completed  : Calculation of chi_L2 map without magnitude weighting.');
+      disp('-----------------------------------------------');
+
+    end
+
+  end
+
+  % DEV note:
+  % L1 flag is raised only if split bregman is selected
+
+  if FitOpt.regL1_Flag && FitOpt.reoptL1_Flag  % || Reopt Lamda L2 case chi_L2 generation
+
+    disp('Started   : Reoptimization of Lamda L1. ...');
+    lambdaL1 = calcSBLambdaL1(phaseLUnwrap, FitOpt.lambdaL1Range, lambdaL2, imageResolution, FitOpt.direction);
+    disp('Completed   : Reoptimization of Lamda L1. ...');
+    disp(['Completed   : Reoptimization of lamda L1. Lambda L1: ' num2str(lambdaL1)]);
+    disp('-----------------------------------------------');
+
+     if isempty(lambdaL1)
+
+        warning('Could not optimize lambda L1 with provided search interval. Setting default value instead.');
+
+        if not(isempty(FitOpt.LambdaL1))
+
+            lambdaL1 = FitOpt.LambdaL1;
+
+        else
+
+            error('Cannot set lambda L1. Please enter a value or change range.')
+
+        end
+    end
+
+  elseif FitOpt.regL1_Flag && not(FitOpt.reoptL1_Flag)
+
+    lambdaL1 = FitOpt.LambdaL1;
+
+  end
+
+
+  if FitOpt.regSB_Flag && FitOpt.magnW_Flag
+
+    disp('Started   : Calculation of chi_SBM map with magnitude weighting.. ...');
+    FitResults.chiSBM = qsmSplitBregman(phaseLUnwrap, maskGlobal, lambdaL1, lambdaL2, FitOpt.direction, imageResolution, padSize, FitOpt.magnW_Flag, magnWeight);
+    disp('Completed   : Calculation of chi_SBM map with magnitude weighting.');
+    disp('-----------------------------------------------');
+
+  elseif FitOpt.regSB_Flag && not(FitOpt.magnW_Flag)
+
+    disp('Started   : Calculation of chi_SBM map without magnitude weighting.. ...');
```

Comment:
@agahkarakuzu Here and in the two lines below, it should be written `SB` and not `SBM` in all instances (text and output).  The 'M' was used in the original scripts for "**M**agnitude-weighted". If you want the fitting variable to be the same name regardless of the magnitude weighted flag value, then I'd change `SBM` to `SB` in the if statement above as well.